src/main/webapp/app/overview/course-exams/course-exams.component.ts

```diff
@@ -47,11 +45,11 @@ export class CourseExamsComponent implements OnInit, OnDestroy {
             this.courseId = parseInt(params['courseId'], 10);
         });
 
-        this.course = this.courseCalculationService.getCourse(this.courseId);
+        this.course = this.courseStorageService.getCourse(this.courseId);
 
-        this.courseUpdatesSubscription = this.courseManagementService.getCourseUpdates(this.courseId).subscribe((course: Course) => {
-            this.courseCalculationService.updateCourse(course);
-            this.course = this.courseCalculationService.getCourse(this.courseId);
+        this.courseUpdatesSubscription = this.courseStorageService.subscribeToCourseUpdates(this.courseId).subscribe((course: Course) => {
```

Comment:
I don't get this code. When I receive an updated course, updateCourse() get's called? And then we again use `getCourse`? I think `this.course = course` should be enough here and the update and notification should be handled from the outside.

Reply:
I think so as well.